lib/crypto/hkdf.lua

```diff
@@ -0,0 +1,115 @@
+--- HMAC-based Extract-and-Expand Key Derivation Function (HKDF) based on RFC 5869.
+-- This module provides functions to perform HKDF operations, utilizing the
+-- underlying `crypto_shash` C module for HMAC calculations.
+--
+-- SPDX-FileCopyrightText: (c) 2025 jperon <cataclop@hotmail.com>
+-- SPDX-License-Identifier: MIT OR GPL-2.0-only
+--
+-- @module crypto.hkdf
+
+local shash = require("crypto_shash")
+local char, sp, rep, sub = string.char, string.pack, string.rep, string.sub
+
+--- HKDF instance methods.
+-- This table represents an HKDF instance configured for a specific hash algorithm.
+-- It is created by calling the main function of the `crypto.hkdf` module.
+-- @type HKDFInstance
+-- @field hkdf Perform the full HKDF (extract and expand).
+-- @field extract Perform the HKDF extract step.
+-- @field expand Perform the HKDF expand step.
+-- @field tls13_expand_label Perform TLS 1.3 HKDF-Expand-Label.
+-- @field close Release resources associated with this HKDF instance.
+
+--- Creates a new HKDF instance for a given hash algorithm.
+-- @tparam string alg The base hash algorithm name (e.g., "sha256", "sha512").
+-- The "hmac(" prefix will be added automatically.
+-- @treturn HKDFInstance An HKDF instance table with methods for key derivation.
+-- @usage local hkdf_sha256 = require("crypto.hkdf")("sha256")
+return function (alg)
```

Comment:
we convention returning tables for lunatik libs.. I would keep the current pattern, unless we have a good reason to change it..